src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1067,7 +1077,11 @@ private IMemberInitializerOperation CreateBoundMemberInitializerOperation(BoundA
         {
             Debug.Assert(IsMemberInitializer(boundAssignmentOperator));
 
-            Lazy<IOperation> target = new Lazy<IOperation>(() => Create(boundAssignmentOperator.Left));
+            Lazy<IOperation> target = new Lazy<IOperation>(() =>
+                boundAssignmentOperator.Left.Kind == BoundKind.ObjectInitializerMember && boundAssignmentOperator.Right.Kind == BoundKind.CollectionInitializerExpression
+                    ? CreateBoundObjectInitializerMemberOperation((BoundObjectInitializerMember)boundAssignmentOperator.Left, inCollectionInitializer: true)
```

Comment:
>CreateBoundObjectInitializerMemberOperation((BoundObjectInitializerMember)boundAssignmentOperator.Left, inCollectionInitializer: true) [](start = 22, length = 134)

Is this call bypassing the cache? Is it possible to end up with two distinct IOperation nodes for the target?

Reply:
It is bypassing the cache, but because we fetch IOperations by getting the enclosing statement and walking down, we shouldn't end up with 2 nodes here.

---
In reply to: [192269300](https://github.com/dotnet/roslyn/pull/27308#discussion_r192269300) [](ancestors = 192269300)